pyentrp/entropy.py

```diff
@@ -102,60 +101,88 @@ def shannon_entropy(time_series):
-
-    for i in range(n - 1):
-        nj = n - i - 1
-        ts1 = time_series[i]
-        for jj in range(nj):
-            j = jj + i + 1
-            if abs(time_series[j] - ts1) < tolerance:  # distance between two vectors
-                curr[jj] = prev[jj] + 1
-                temp_ts_length = min(sample_length, curr[jj])
-                for m in range(int(temp_ts_length)):
-                    A[m] += 1
-                    if j < n - 1:
-                        B[m] += 1
-            else:
-                curr[jj] = 0
-        for j in range(nj):
-            prev[j] = curr[j]
-
-    N = n * (n - 1) / 2
-    B = np.vstack(([N], B[:sample_length - 1]))
-    similarity_ratio = A / B
-    se = - np.log(similarity_ratio)
-    se = np.reshape(se, -1)
-    return se
-
-
-def multiscale_entropy(time_series, sample_length, tolerance):
+    Ntemp = 0.0;
+    Ntemp_plus = 0.0;
+    
+    for i in range(n-m-1):
+        template = time_series[i:(i+m+1)];#We have 'm+1' elements in the template
+        rem_time_series = time_series[i+1:]
+        
+        """Search for matches in remtime_series. First we look for match with template[0]. 
+        We then select the neighbors of these that also match next elements """
+        
+        "Create a list with those indices in the time series that match the first element in template"
+        searchlist = np.nonzero(np.abs(rem_time_series - template[0]) < tol)[0]
+
+        go = len(searchlist) > 0;
+        
+        length = 1;
+       
+        if length == m:
+            Ntemp += len(searchlist)
+        
+        """This while loop keeps reducing the searchlist by also comparing next elements making the templtates longer. 
+        It will stop if there are no elements that match or if we reach m+1-length-templates. """
+        while go:
+            
+            "Shift the index 1 step to the right in order to find elements in the time series next to the ones we found:"
+            nextindxlist = searchlist + 1;
+            "remove elements to close to the end:"
+            nextindxlist = nextindxlist[nextindxlist < n - 1 - i]
+           
+            "This is the list of elements that we shopuld compare with the next entry in the template vector:"
+            nextcandidates = rem_time_series[nextindxlist]
+            
+            "Hitlist is bool list and true where next time_series elements match next template elements"
+            hitlist = np.abs(nextcandidates - template[length]) < tol
+            "reduce the search list to those elements that surviced this round"
+            searchlist = nextindxlist[hitlist]
+           
+            if length == m-1:
+                Ntemp += np.sum(hitlist)
+            elif length == m :
+                Ntemp_plus += np.sum(hitlist)
+            
+           
+            length += 1;
+            go = any(hitlist) and length < m + 1    
+            
+            
+    #print('Ntemp m', Ntemp)
```

Comment:
Let's remove it? Seem like it's not necessary to keep comments

Reply:
Ok.